SMTPServer/MailSession.cpp

```diff
@@ -0,0 +1,299 @@
+// This is an independent project of an individual developer. Dear PVS-Studio, please check it.
+
+// PVS-Studio Static Code Analyzer for C, C++, C#, and Java: http://www.viva64.com
+
+#include "MailSession.h"
+
+MailSession::MailSession(SOCKET& client_socket)
+{
+	this->m_client_socket = client_socket;
+}
+
+const SOCKET& MailSession::GetSocket() const
+{
+	return m_client_socket;
+}
+
+bool MailSession::ValidAdress(char* buf)
+{
+	size_t strlen_buf = strlen(buf);
+	return (strlen_buf > 2 && strlen_buf < 255 && strchr(buf, '@'));
+}
+
+std::string MailSession::CutAddress(char* buf)
+{
+	std::string str_buf = buf;
+	std::string result = "";
+
+	size_t start_pos = str_buf.find('<', 0);
+	size_t end_pos = str_buf.find('>', 0);
+
+	for (size_t i = start_pos + 1; i < end_pos; i++)
+	{
+		result += str_buf.at(i);
+	}
+
+	return result;
+}
+
+std::string MailSession::CutSubject(char* buf)
+{
+	std::string str_buf = buf;
+	std::string result = "";
```

Comment:
String initialization.

Reply:
I can't use "append()". It calls an exception